addons/hr_evaluation/models/hr_evaluation.py

```diff
@@ -0,0 +1,344 @@
+# -*- coding: utf-8 -*-
+import calendar
+import datetime
+import uuid
+from dateutil.relativedelta import relativedelta
+
+from openerp import api, fields, models, _
+from openerp.exceptions import UserError
+
+
+class HrEvaluation(models.Model):
+    _name = "hr_evaluation.evaluation"
+    _inherit = ['mail.thread']
+    _description = "Employee Appraisal"
+    _order = 'date_close, interview_deadline'
+
+    EVALUATION_STATE = [
+        ('new', 'To Start'),
+        ('pending', 'Appraisal Sent'),
+        ('done', 'Done')
+    ]
+
+    @api.multi
+    def _track_subtype(self, init_values):
+        self.ensure_one()
+        if 'state' in init_values and self.state == 'new':
+            return 'hr_evaluation.mt_appraisal_new'
+        if 'interview_deadline' in init_values and not self.meeting_id or self.env.context.get('meeting'):
+            return 'hr_evaluation.mt_appraisal_meeting'
+        return super(HrEvaluation, self)._track_subtype(init_values)
+
+    @api.one
+    def _compute_users_input(self):
+        survey_ids = [
+            self.appraisal_manager_survey_id.id,
+            self.appraisal_colleagues_survey_id.id,
+            self.appraisal_self_survey_id.id,
+            self.appraisal_subordinates_survey_id.id
+        ]
+        self.user_input_ids = self.env['survey.user_input'].search([
+            ('survey_id', 'in', survey_ids),
+            ('type', '=', 'link'), ('survey_res_id', '=', self.id), ('survey_model', '=', self._name)])
+
+    @api.one
+    def _compute_user_input_count(self):
+        self.user_input_count = len(self.user_input_ids)
+
+    @api.one
+    def _compute_completed_user_input(self):
```

Comment:
@tde-banana-odoo , we should keep compute o2m field. we does not added link of evaluation in survey.user.input model. so we need to search records of survey user inputs which has survey_model='hr.evaluation' and res_id=current record of evaluation.

we added two fields: survey_model and res_id in survey.user.input to (loose) bind any model with survey.user.input record.